src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -206,16 +269,25 @@ bool _applicableFilter(Index *idx, OpFilter *filter) {
 	// Make sure the filter root is not a function.
 	// TODO: As a result of issue 667, transform "IN" into a sequence of "OR" to use in RedisSearch.
 	if(filter_tree->t == FT_N_EXP) {
-		res = false;
```

Comment:
As we only allow `FT_N_EXP` to represent an `in` expression, move `if(!strcasecmp(exp->op.func_name, "in") == 0) return false;` from `_validateInExpression` here.

Reply:
done